lightning/src/ln/onion_utils.rs

```diff
@@ -2413,6 +2413,94 @@ mod tests {
 		assert_eq!(decrypted_failure.onion_error_code, Some(0x2002));
 	}
 
+	#[test]
+	fn test_trampoline_onion_error_vectors() {
```

Comment:
I think this commit can be squashed with the previous one. It is a logical unit of change, at least for me.